core/src/main/java/tachyon/worker/netty/BlockResponse.java

```diff
@@ -70,28 +68,19 @@ protected void encode(final ChannelHandlerContext ctx, final BlockResponse msg,
       out.add(createHeader(ctx, msg));
 
       final BlockHandler handler = msg.getHandler();
-      if (handler == null) {
-        return;
-      }
-      final FileTransferType type =
-          mTachyonConf
-              .getEnum(Constants.WORKER_NETTY_FILE_TRANSFER_TYPE, FileTransferType.TRANSFER);
-      switch (type) {
-        case MAPPED:
-          ByteBuffer data = handler.read(msg.getOffset(), (int) msg.getLength());
-          out.add(Unpooled.wrappedBuffer(data));
-          handler.close();
-          break;
-        case TRANSFER: // intend to fall through as TRANSFER is the default type.
-        default:
-          if (handler.getChannel() instanceof FileChannel) {
-            out.add(new DefaultFileRegion((FileChannel) handler.getChannel(), msg.getOffset(),
-                msg.getLength()));
-          } else {
-            handler.close();
-            throw new Exception("Only FileChannel is supported!");
-          }
-          break;
+      if (handler != null) {
+        final FileTransferType type =
+            mTachyonConf
+                .getEnum(Constants.WORKER_NETTY_FILE_TRANSFER_TYPE, FileTransferType.TRANSFER);
+        switch (type) {
+          case MAPPED:
+            ByteBuffer buffer = handler.read(msg.getOffset(), (int)msg.getLength());
+            out.add(Unpooled.wrappedBuffer(buffer));
+            break;
+          default: // TRANSFER
```

Comment:
I think there was a clean up PR including `case TRANSFER` to be explicit. Is there any downside to including the case and just having it fall through?